plugins/precise-volume/menu.js

```diff
@@ -1,19 +1,109 @@
 const { enabled } = require("./back");
+const { app } = require("electron");
 const { setOptions } = require("../../config/plugins");
+const prompt = require("custom-electron-prompt");
+const path = require("path");
+const is = require("electron-is");
 
 module.exports = (win, options) => [
 	{
-		label: "Arrowkeys controls",
+		label: "Local Arrowkeys Controls",
 		type: "checkbox",
 		checked: !!options.arrowsShortcut,
-		click: (item) => {
-			// Dynamically change setting if plugin enabled
+		click: item => {
+			// Dynamically change setting if plugin is enabled
 			if (enabled()) {
 				win.webContents.send("setArrowsShortcut", item.checked);
 			} else { // Fallback to usual method if disabled
 				options.arrowsShortcut = item.checked;
 				setOptions("precise-volume", options);
 			}
 		}
+	},
+	{
+		label: "Global Hotkeys",
+		type: "checkbox",
+		checked: !!options.globalShortcuts.volumeUp || !!options.globalShortcuts.volumeDown,
+		click: item => promptGlobalShortcuts(win, options, item)
+	},
+	{
+		label: "Set Custom Volume Steps",
+		click: () => promptVolumeSteps(win, options)
 	}
 ];
+
+const iconPath = path.join(app.getAppPath(), "assets", "youtube-music-tray.png");
+const customTitlebarPath = path.join(app.getAppPath(), "plugins", "in-app-menu", "prompt-custom-titlebar.js");
+// Helper function for globalShortcuts prompt
+const kb = (label_, value_, default_) => { return { value: value_, label: label_, default: default_ || undefined }; };
+
+function promptVolumeSteps(win, options) {
+	const promptOptions = {
+		title: "Volume Steps",
+		label: "Choose Volume Increase/Decrease Steps",
+		value: options.steps || 1,
+		type: "counter",
+		counterOptions: { minimum: 0, maximum: 100, multiFire: true },
+		width: 380
+	};
+
+	setupPromptOptions(promptOptions);
+
+	prompt(promptOptions, win)
+		.then(output => {
+			if (output || output === 0) { // 0 is somehow valid
+				options.steps = output;
+				setOptions("precise-volume", options);
+			}
+		}).catch(console.error);
+}
+
+function promptGlobalShortcuts(win, options, item) {
+	const promptOptions = {
+		title: "Global Volume Keybinds",
+		label: "Choose Global Volume Keybinds:",
+		type: "keybind",
+		keybindOptions: [
+			kb("Increase Volume", "volumeUp", options.globalShortcuts?.volumeUp),
+			kb("Decrease Volume", "volumeDown", options.globalShortcuts?.volumeDown)
+		]
+	};
+
+	setupPromptOptions(promptOptions);
+
+	prompt(promptOptions, win)
+		.then(output => {
+			if (output) {
+				for (const keybindObject of output) {
+					options.globalShortcuts[keybindObject.value] = keybindObject.accelerator;
```

Comment:
Similarly, doing it by destructuring options is easier to read (and does not require checking `if (output)`):
```suggestion
				options = {
				  ...options,
				  globalShortcuts: {
				     ...options.globalShortcuts,
				     ...output
				}   
```

Reply:
But output is in format (value: "volumeUp", accelerator: "Shift+A")
which isn't the same as options.globalShortcuts (volumeUp: "Shift+A")